modules/update-cluster-version-object.adoc

```diff
@@ -0,0 +1,116 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-cluster-version-object_{context}"]
+= The ClusterVersion object
+
+One of the resources that the Cluster Version Operator (CVO) monitors is `ClusterVersion`.
+
+`ClusterVersion` is a custom resource object that contains information relating to the cluster's version, such as the current and desired versions of the cluster.
+When the CVO observes that the desired version does not match the current version in the `ClusterVersion` resource, it attempts to initiate an update to reconcile the cluster with this new desired state.
+
+
+//to-do: this might be heading overload, consider deleting this heading if the context switch from the previous paragraph to this content is smooth enough to not require one.
```

Comment:
one way to  structure would be a generic "consume `spec`, reconcile, report in `status`"  to underline how we match [the usual Kubernetes pattern][1].  Another way to structure would be to have a few sections:

* Reconciling the currently accepted target release, which effectively consumes `status.desired` and reports via `Progressing`, `Failing`, etc.
* Providing next-hop advice, which consumes `spec.upstream` and `channel` and reports in `status.availableUpdates` and `conditionalUpdates` and `RetrievedUpdates`.
  * Also in this line, we're consuming ClusterOperator `Upgradeable` and producing ClusterVersion `Upgradeable`.
* Accepting a proposed next hop, which consumes `spec.desiredUpdate`, `status.availableUpdates`, `conditionalUpdates`, `Upgradeable`, and release signatures, and reports in `status.desired` and `RetrievePayload`.

Each of those touches up against admin activity.  During an update, an admin will bump up against all of those controller loops.  Outside of updates, admins will mostly care if  there are issues reconciling the currently accepted target release, until they start planning and preparing for their next round of updates.

I'm personally agnostic about whether it's easier to explain ClusterVersion as a generic Kube `spec`/`status` resource that happens to be about cluster reconciliation and updates, or if it's easier to explain ClusterVersion as interacting with a series of controllers broken down by use-case.

[1]: https://kubernetes.io/docs/concepts/overview/working-with-objects/#object-spec-and-status